### ISSUE TITLE
Mock Lambda event server gets stuck on responses with large response sizes when using Jackson

### ISSUE THREAD

**Issue by c-classen** [quarkusio/quarkus#44551]
### Describe the bug

When using the Lambda event server and requesting data from an endpoint, the HTTP response gets stuck after sending about 10KB  for large responses.

### Expected behavior

10 KB should be not issue. AWS Lambda supports up to 10 MB of response size.

### Actual behavior

The response to the request never gets completed

### How to Reproduce?

Reproducer: [lambda-gets-stuck.zip](https://github.com/user-attachments/files/17791040/lambda-gets-stuck.zip)

or

1. Go to https://code.quarkus.io/ and add `AWS Lambda HTTP` and `REST Jackson`.
2. Replace the content of the `hello()` method in `GreetingResource` with 
    ````java
    List<String> greetings = new ArrayList<>();
    for (int i = 0; i < 580; i++) {
        greetings.add("Hello World " + i);
    }
    return greetings;
3. Start the dev server and run `curl localhost:8080/hello`

It should just get stuck and never finish, although a lot of the response is displayed.


### Output of `uname -a` or `ver`

Linux clemens-WAPP-Laptop 6.8.0-45-generic #45~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Sep 11 15:25:05 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "21.0.2" 2024-01-16 LTS OpenJDK Runtime Environment Corretto-21.0.2.14.1 (build 21.0.2+14-LTS) OpenJDK 64-Bit Server VM Corretto-21.0.2.14.1 (build 21.0.2+14-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.16.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

wrapperVersion=3.3.2

### Additional information

It seems to me that the following factors need to come together to provoke this behavior:

1. **Running the dev server**: When deploying to AWS, it seems to work
2. **Large response size:**. Interestingly, reducing `580` to `570` will get it stuck earlier, but with `200`, it works for me. Since there might be timing issues at play, this could vary and you might need to increase 580 to a sufficiently high number.
3. **JSON serialization**: Adding `@Produces(MediaType.TEXT_PLAIN)` causes the bug to disappear, even when using much higher upper bounds in the for loop such as `10_000`.
4. **Lambda HTTP Dependency**: Commenting out the `quarkus-amazon-lambda-http` dependency in the `pom.xml` will also make the bug disappear.

**Comment by c-classen** [comment]
@patriot1burke It would already help a lot if there was a configuration option to disable the mock Lambda event server when running the application in dev mode without having to remove the dependency to AWS Lambda HTTP.

**Comment by andrezimmermann** [comment]
If the type of response is set to something like `@Produces(MediaType.TEXT_PLAIN)` 

```java
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public List<String> hello() {
        List<String> greetings = new ArrayList<>();
        for (int i = 0; i < 580; i++) {
            greetings.add("Hello World " + i);
        }
        return greetings;
    }
```

It will run correctly

![Image](https://github.com/user-attachments/assets/c9a92639-cfe4-473e-87e0-3ee10b9d9cda)


I believe if there is no explicit content type, Quarkus will see the List<> and default to a Server Side Event response.
https://quarkus.io/guides/rest#server-sent-event-sse-support

Not sure how the  Quarkus AWS Lambda HTTP implements it, but you need set the AWS Lambda itself as a "Streaming" response.
https://aws.amazon.com/blogs/compute/introducing-aws-lambda-response-streaming/ 
I'd assume that this feature is out of the scope of the `Quarkus AWS Lambda HTTP`

In any case, as good practice always annotate the controller's response types with `@Produces` to avoid the wrong default being used.

**Comment by c-classen** [comment]
@andrezimmermann Thank you for your help. I tried setting `@Produces(MediaType.TEXT_PLAIN)` and it did indeed complete the response without getting stuck.

However, I still want my endpoint to return JSON (and also have the spec generate accordingly and the `Content-Type` header set), so I would need to use `@Produces(MediaType.APPLICATION_JSON)`. And when I set this instead, I still run into the same issue, even though a default to `@Produces(MediaType.SERVER_SENT_EVENTS)` as mentioned in the first link would not make sense to me when explicitly specifying `@Produces(MediaType.APPLICATION_JSON)`.

Regarding the streaming response setting, please consider that the code runs fine on the actual AWS Lambda. It is just the Quarkus dev mode run locally that has the issue.

**Comment by geoand** [comment]
Is there a chance you can [check with Quarkus main](https://github.com/quarkusio/quarkus/blob/main/CONTRIBUTING.md#checking-an-issue-is-fixed-in-main) and see if the problem persists?
Asking because we made a fix to another issue that might be related to this.

**Comment by geoand** [comment]
@c-classen any chance you can check the above comment?

**Comment by c-classen** [comment]
@geoand Sorry, I had a lot to do and kind of forgot about this. I used the snapshot described in the link you provided and within my reproducer application, the issue stopped appearing, so thank you for fixing it!

Also thanks for providing the nightly snapshots and the guide to try them out. That was a lot easier than building Quarkus myself. Maybe you could put in a little hint that the `settings.xml` should go in the `.m2` directory in the home directory. I know it is not hard to google, but it would just make it more convenient for people like me who usually only use Gradle.

**Comment by c-classen** [comment]
I'm not sure whether I made some mistake when checking with Quarkus main last time or whether this is a regression, but the problem is currently present in the latest Quarkus main.

To confirm it, I downloaded my reproducer again, changed the `quarkus.platform.version` to `999-SNAPSHOT`, started the application and ran the curl. It always seems to get stuck right before the response would have finished.

I found this while updating one of our applications to Quarkus 3.20.0 where the problem also exists.

**Comment by geoand** [comment]
Is the problem still reproducible with the attached project?

**Comment by c-classen** [comment]
Yes, it is right now. I tried it again a few minutes ago with `999-SNAPSHOT` in `quarkus.platform.version` and it got stuck again at "Hello World 574".

I packed it again with the `999-SNAPSHOT` property: 

[lambda-gets-stuck-999-snapshot.zip](https://github.com/user-attachments/files/20337766/lambda-gets-stuck-999-snapshot.zip)

You can just download it, unzip it, run `./mvnw quarkus:dev` and then in another terminal `curl localhost:8080/hello` to see the issue.
